python/ray/util/sgd/tests/test_torch_runner.py

```diff
@@ -203,54 +203,96 @@ def _testWithInitialized(self, init_mock):
             self.assertIn(env_set_device, visible_devices)
             device_int = int(local_device)
             self.assertLess(device_int, len(visible_devices))
+            self.assertEquals(len(os.environ["CUDA_VISIBLE_DEVICES"]), 1)
         else:
             self.assertEquals(local_device, env_set_device)
 
     def testNoVisibleWithInitialized(self):
         with patch("torch.cuda.is_initialized") as init_mock:
             init_mock.return_value = True
-            self._testWithInitialized(init_mock)
+            self._testReserveCUDAResource(init_mock, 0)
 
-    def test2VisibleWithInitialized(self):
-        os.environ["CUDA_VISIBLE_DEVICES"] = "2,3"
+    def testNoVisibleWithInitializedAndReserveCPUResource(self):
         with patch("torch.cuda.is_initialized") as init_mock:
             init_mock.return_value = True
-            self._testWithInitialized(init_mock)
+            self._testReserveCUDAResource(init_mock, 2)
 
     def test1VisibleWithInitialized(self):
         os.environ["CUDA_VISIBLE_DEVICES"] = "0"
         with patch("torch.cuda.is_initialized") as init_mock:
             init_mock.return_value = True
-            self._testWithInitialized(init_mock)
+            self._testReserveCUDAResource(init_mock, 0)
 
-    def test2VisibleNotInitialized(self):
+    def test1VisibleWithInitializedAndReserveCPUResource(self):
+        os.environ["CUDA_VISIBLE_DEVICES"] = "0"
+        with patch("torch.cuda.is_initialized") as init_mock:
+            init_mock.return_value = True
+            self._testReserveCUDAResource(init_mock, 2)
```

Comment:
Also don't think we need this test, it seems redundant

Reply:
removed